lib/mix/test/mix/cli_test.exs

```diff
@@ -81,18 +81,18 @@ defmodule Mix.CLITest do
     end
   end
 
-  test "--help smoke test", context do
+  test "-h smoke test", context do
```

Comment:
We should try to test the CLI module directly instead since adding more smoke tests will make the suite slower. They are by far the most time consuming part of the suite.